lib/events/dynamoevents/dynamoevents.go

```diff
@@ -1235,6 +1235,29 @@ func (l *Log) convertFieldsToDynamoMapFormat(ctx context.Context) error {
 	return trace.Wrap(l.migrateMatchingEvents(ctx, filterExpr, transformEvent))
 }
 
+func minInt64(x, y int64) int64 {
+	if x > y {
+		return y
+	}
+	return x
+}
+
+func (l *Log) approximateOptimalMigrationWorkers() (int64, error) {
+	req := dynamodb.DescribeTableInput{TableName: aws.String(l.Tablename)}
+	table, err := l.svc.DescribeTable(&req)
+	if err != nil {
+		return 0, trace.Wrap(err)
+	}
+
+	// calculate the throughput, accounting for r/w bottlenecks
+	provisioned := table.Table.ProvisionedThroughput
+	throughput := minInt64(*provisioned.ReadCapacityUnits, *provisioned.WriteCapacityUnits)
+
+	// divide throughput by batch size rounding upwards and then take 75% of that
+	optimalWorkers := (throughput + (DynamoBatchSize - 1)) / DynamoBatchSize / 4 * 3
```

Comment:
I don't know how large we expect these numbers to be but doing the divisions last could make it less likely this will round to 0 at an intermediate value. Should we make sure it minimally returns 1 as well?

Reply:
Fixed, clamped to 1 at minimum.